nsqadmin/static/js/models/node.js

```diff
@@ -0,0 +1,25 @@
+var $ = require('jquery');
+
+var AppState = require('../app_state');
+var Backbone = require('backbone');
+
+var Node = Backbone.Model.extend({
+    idAttribute: 'name',
+
+    constructor: function Node() {
+        Backbone.Model.prototype.constructor.apply(this, arguments);
+    },
+
+    url: function() {
+        return AppState.url('/nodes/' + this.get('name'));
```

Comment:
I think you can just define
```
urlRoot: function() { 
    return AppState.url('/nodes');
}
```
to make this work because you have `idAttribute: 'name'` already set.

http://backbonejs.org/#Model-urlRoot
and the code:
http://backbonejs.org/docs/backbone.html#section-89

But maybe that's too much magic/abstraction?